README.md

```diff
@@ -408,7 +408,8 @@ The arguments for *compile* are:
 * `--config` to inspect the runtime compile configuration (see below).
 * `-S` or `--supported` shows a matrix of the supported targets and toolchains.
 * `-f` or `--flash` to flash/program a connected target after successful compile.
-* `-c ` to build from scratch, a clean build or rebuild.
+* `-s` or `--sterm` to open the connected target's serial terminal. This option can be chained with `-f` or `-r` so the new firmware image will be flashed to the target or reset, then open the serial terminal.
```

Comment:
-s and -S seem confusing. I get -fr for flash and reset but seems that ending in a terminal should be an explicit --sterm or better --terminal or --term

Reply:
We would have to change the code than handles arguments to make that change. That's out of scope for this PR, probably.

Reply:
I personally don't like using the word "terminal" here. It's a tad off, as it's specifically a hardware thing https://en.wikipedia.org/wiki/Computer_terminal . What were talking about is really a terminal emulator. We could also claim that daplink is the terminal, and that we are connecting to it, so `--connect` might also make sense.

Reply:
@theotherjimmy In terms of other tutorials (like the [offline quickstart](https://os.mbed.com/docs/v5.8/tutorials/quick-start-offline.html#printf)) we refer to viewing serial port output as opening up a "serial terminal". I personally think calling it an emulator is confusing. Thoughts? Maybe "serial port viewer"?

Reply:
In terms of how the word "terminal" is used these days, it probably means "something that takes input, accepts output and looks like a command line". That's why I mentioned that it was a personal pref. Also, calling it an "emulator" is confusing because you omit "terminal" and the user is left without the context of what is being emulated.

-------------

I think that "serial port viewer" may have the same problem that "sterm" and "serial terminal" has: it starts with an "s", making the short option `-s`, which conflicts with `-S`. Further, I would rather we use terminal here, as it's a better understood word for this than "port viewer". Especially as "port viewer" might imply to a user that they cannot interact with the device.

Reply:
Agreed with @yennster. Serial terminal is rather established term (check mini term).

Terminal emulator is very confusing - we're no longer in the 80's/90s when a terminal meant a monitor and a keyboard attached to a system, usually part of a bigger system.

"connect" doesn't make sense either, especially in the light of terminology in the IoT world.

Reply:
https://en.wikipedia.org/wiki/Terminal_mode

Reply:
@screamerbg I claimed that it was a personal pref.

-----

That Terminal mode article links to "Terminal Emulator" and makes no assertion that a Terminal is not a hardware device, or that a Terminal Emulator would not be capable of emulating the same behavior.

-----

I think that "serial terminal" is inconsistent with `compile`, `detect`, `--flash` and  `--reset`, as it's a noun instead of a verb. I proposed "connect" because it fit the pattern of verbs.

Reply:
@sg- Agreed that `-s` and `-S` are somewhat confusing. We can speculate which one would be more used `-S` (for supported) or `-s` for serial terminal, yet for backwards compatibility we can't deprecate `-S`

Perhaps we could rethink the workflow itself. For example compile+flash+term can be chained via `mbed compile -f --term` (which is fine with me), but for flash+terminal only a user could use `mbed term -f` instead of `mbed detect -f -s`. E.g. introduce verb `mbed term` with all required options, like baudrate, encoding, etc. instead of cramming it in `mbed detect`.

What do you think?

Reply:
I'm on board with adding another verb for this extended workflow. We might have contention about naming that verb though. `term` seems like a noun to me.

Reply:
`mbed terminal`?

Note that `mbed status` is not a verb, nor is `mbed releases`

Reply:
Well you have me there :D The `-s` vs `-S` conflict exists on `mbed compile` only, as far as I know. Are you suggesting that `mbed term`/`mbed terminal` builds the code too? Maybe through a switch? Maybe as opt-out?

Reply:
`mbed monitor`? 😆

Reply:
Is "console" the word you're looking for?

Reply:
`mbed console` sounds good, but then the option `-c` would conflict with the exisiting `-c` for a clean build